src/StyleSheet.js

```diff
@@ -89,7 +89,7 @@ export default class StyleSheet {
       if (!this.deployed) return rule
       // Don't insert rule directly if there is no stringified version yet.
       // It will be inserted all together when .attach is called.
-      if (queue) queue.push(rule)
+      if (queue) queue.unshift(rule)
```

Comment:
I am not sure why this is not breaking anything else. You are reverting the order of added rules for all addRule calls. Maybe there is a missing test which ensures the right order.

We have actually an `options.index` which is set in jss-nested in order to have the control over the order. Maybe jss-nested sets the wrong index?